x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/EsqlUnresolvedRelation.java

```diff
@@ -17,22 +18,34 @@
 public class EsqlUnresolvedRelation extends UnresolvedRelation {
 
     private final List<Attribute> metadataFields;
-
-    public EsqlUnresolvedRelation(Source source, TableIdentifier table, List<Attribute> metadataFields, String unresolvedMessage) {
+    private final IndexMode indexMode;
+
+    public EsqlUnresolvedRelation(
```

Comment:
Consider providing two factory methods setting the mode, e.g. `create()` and `createTsdb()`